pkg/controller/image-stream-tag-reconciler/reconciler.go

```diff
@@ -46,21 +56,43 @@ func AddToManager(mgr controllerruntime.Manager, opts Options) error {
 		return fmt.Errorf("failed to add prowv1 to scheme: %w", err)
 	}
 
+	if err := opts.CIOperatorConfigAgent.AddIndex(configIndexName, configIndexFn); err != nil {
+		return fmt.Errorf("failed to add indexer to config-agent: %w", err)
+	}
+
+	createdJobsCounter := prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: controllerName,
+		Name:      "prowjobs_created",
+		Help:      "The number of prowjobs the controller created",
+	}, []string{"org", "repo", "branch", "success"})
+	if err := metrics.Registry.Register(createdJobsCounter); err != nil {
+		return fmt.Errorf("failed to register createdJobsCounter: %w", err)
+	}
+
+	log := logrus.WithField("controller", controllerName)
 	r := &reconciler{
-		ctx:              context.Background(),
-		log:              logrus.WithField("controller", "imageStreamTagReconciler"),
-		client:           mgr.GetClient(),
-		dryRun:           opts.DryRun,
-		ciOPConfigAgent:  opts.CIOperatorConfigAgent,
-		prowJobNamespace: opts.ProwJobNamespace,
-		gitClient:        opts.GitClient,
-	}
-	c, err := controller.New(
-		"imageStreamTagReconciler",
-		mgr,
+		ctx:                 context.Background(),
+		log:                 log,
+		client:              imagestreamtagwrapper.New(mgr.GetClient()),
+		releaseBuildConfigs: opts.CIOperatorConfigAgent,
+		dryRun:              opts.DryRun,
+		prowJobNamespace:    opts.ProwJobNamespace,
+		gitClient:           opts.GitClient,
+		createdProwJobLabels: map[string]string{
+			"openshift.io/created-by": controllerName,
+		},
+		ignoredGitHubOrganizations: sets.NewString(opts.IgnoredGitHubOrganizations...),
+		createdJobsCounter:         createdJobsCounter,
+	}
+	c, err := controller.New(controllerName, mgr, controller.Options{
+		// Since we watch ImageStreams and not ImageStreamTags as the latter do not support
+		// watch, we create a lot more events the needed. In order to decrease load, we coalesce
+		// and delay all requests after a successful reconciliation for up to an hour.
+		Reconciler: controllerutil.NewReconcileRequestCoalescer(r, time.Hour),
```

Comment:
I think we should look at the mean time taken to run a publication job and use that as the delay here, if we can -- does a change from 1h to 10m make a significant change in throughput?

Reply:
Set it to 10mins, now that we use a GitHub client the worqueue depth goes down to zero after a minute or so